src/NuGet.Core/NuGet.Packaging/Signing/Verification/NuGetSignatureWhitelistObjectType.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Packaging.Signing
+{
+    public enum NuGetSignatureWhitelistObjectType
```

Comment:
I agree that it could be useful to restrict which signatures/countersignatures to which validation applies.

Perhaps a more flexible, generalized enum is as follows.

```
[Flags]
public enum SignatureVerificationTarget
{
    Primary = 1,
    Repository = 2
}
```

I'm not crazy about the type name but I couldn't think of a better one in the moment.  [`SignatureType`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Signing/Signatures/SignatureType.cs) describes the signature type (via `commitment-type-indication` attribute usage), and [`SignaturePlacement`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Signing/Signatures/SignaturePlacement.cs) describes the position of the signature (either the primary signature or a countersignature of the primary signature).